gadcp/madcp.py

```diff
@@ -403,11 +403,24 @@ def _parse_meta_data(self):
 
         """
         essential_meta_data = ["lon", "lat"]
+
         [
             self._safely_add_attribute_from_params(k, self.meta_data)
             for k in essential_meta_data
         ]
 
+        # Check SN
+        sn_internal = int.from_bytes(self.tsdat.FL.Inst_SN, "little")
+
+        if "sn" not in self.meta_data:
+            self.meta_data.sn = sn_internal
+
+        # Check internal SN matches user set one
+        if sn_internal != self.meta_data.sn:
+            warn(
```

Comment:
I am trying not to have many warnings in the code but rather pipe them to the logger and then let the user decide what to do with it. In this case it seems to be good to warn the user no matter what they want to see since it seems pretty severe to have diverging SNs. I may add a statement that also writes this to the log file.

Reply:
Hmm, I guess the logger is not set up at this point. Okay, maybe good enough to spit out the warning to the terminal.